benchmark/client.go

```diff
@@ -0,0 +1,235 @@
+package main
+
+import (
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/GaryBoone/GoStats/stats"
+
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+)
+
+type SubTimes map[string][]float64
+
+// Client - represents mqtt client
+type Client struct {
+	ID         string
+	BrokerURL  string
+	BrokerUser string
+	BrokerPass string
+	MsgTopic   string
+	MsgSize    int
+	MsgCount   int
+	MsgQoS     byte
+	Quiet      bool
+	mqttClient *mqtt.Client
+	Mtls       bool
+	SkipTlsVer bool
+	CA         []byte
+	clientCert tls.Certificate
+	ClientKey  *rsa.PrivateKey
+}
+
+// RunPublisher - runs publisher
+func (c *Client) RunPublisher(res chan *RunResults, mtls bool) {
+	newMsgs := make(chan *Message)
```

Comment:
Can you put some buffer size here, so that you get better utilization of goroutines?